pkg/controller/resourceclaim/controller.go

```diff
@@ -314,82 +336,179 @@ func (ec *Controller) syncPod(ctx context.Context, namespace, name string) error
 		return nil
 	}
 
+	var newPodClaims map[string]string
 	for _, podClaim := range pod.Spec.ResourceClaims {
-		if err := ec.handleClaim(ctx, pod, podClaim); err != nil {
+		if err := ec.handleClaim(ctx, pod, podClaim, &newPodClaims); err != nil {
 			if ec.recorder != nil {
 				ec.recorder.Event(pod, v1.EventTypeWarning, "FailedResourceClaimCreation", fmt.Sprintf("PodResourceClaim %s: %v", podClaim.Name, err))
 			}
 			return fmt.Errorf("pod %s/%s, PodResourceClaim %s: %v", namespace, name, podClaim.Name, err)
 		}
 	}
 
+	if newPodClaims != nil {
```

Comment:
this means we're now doing two writes in a single run of the control loop (create of the resourceclaim and update of the status)... doesn't this introduce new error cases, like:
* resourceclaim creation succeeds, pod status update fails, resourceclaim gets orphaned
* controller resourceclaim cache doesn't yet contain a previously created resourceclaim object for the pod, so we double create with a new generated name

Reply:
I've added one commit which adds a `MutationCache`. Wasn't too complex and thus probably is worth doing.